common/src-mapred/org/pentaho/hadoop/mapreduce/GenericTransReduce.java

```diff
@@ -47,329 +40,260 @@
-    return reduceOutputStepName;
-  }
-
-  public void reduce(final K key, final Iterator<V> values, final OutputCollector<K2, V2> output, final Reporter reporter) throws IOException {
-    try {
-      if (debug) reporter.setStatus("Begin processing record");
-
-      // Just to make sure the configuration is not broken...
-      //
-      if (trans == null) {
-        throw new RuntimeException("Error initializing transformation.  See error log."); //$NON-NLS-1$
-      }
-      
-      // If we're using the single threading engine we're going to keep pushing rows into our construct.
-      // If not, we're going to re-create the Trans engine every time.
-      //
-      if (isSingleThreaded()) {
-        
-        // Only ever initialize once!
-        //
-        if (!trans.isRunning()) {
-          // The transformation needs to be prepared and started...
-          // 
-          shareVariableSpaceWithTrans(reporter);
-          setTransLogLevel(reporter);
-          prepareExecution(reporter);
-          addInjectorAndProducerToTrans(key, values, output, reporter, getInputStepName(), getOutputStepName());
-          
-          // Create the Single Threader executor, sort the steps, and so on...
-          //
-          executor = new SingleThreadedTransExecutor(trans);
-          
-          // This validates whether or not a step is capable of running in Single Threaded mode.
-          //
-          boolean ok = executor.init();
-          if (!ok) {
-            throw new KettleException("Unable to initialize the single threaded transformation, check the log for details.");
-          }
-          
-          // The transformation is considered in a "running" state now.
-        } 
-        
-        // The following 2 statements are the only things left to do for one set of data coming from Hadoop...
-        //
-        
-        // Inject the values, including the one we probed...
-        //
-        injectValues(key, values, output, reporter);
-        
-        // Signal to the executor that we have enough data in the pipeline to do one iteration.
-        // All steps are executed in a loop once in sequence, one after the other.
-        //
-        executor.oneIteration();
-
-      } else {
-        
-        // Clean up old logging
-        //
-        KettleLogStore.discardLines(trans.getLogChannelId(), true);
-        
-        // Create a copy of trans so we don't continue to add new TransListeners and run into a ConcurrentModificationException
-        // when this reducer is reused "quickly"
-
-        trans = MRUtil.recreateTrans(trans);
-        
-        shareVariableSpaceWithTrans(reporter);
-        setTransLogLevel(reporter);
-        prepareExecution(reporter);
-        addInjectorAndProducerToTrans(key, values, output, reporter, getInputStepName(), getOutputStepName());
+        Reducer<K, V, K2, V2> {
+
+    private static Logger logger = Logger.getLogger(GenericTransReduce.class);
```

Comment:
This variable (and thus its import) don't appear to be used and are likely not needed/desired